src/KDSoapServer/KDSoapServerSocket.cpp

```diff
@@ -444,7 +558,123 @@ bool KDSoapServerSocket::handleFileDownload(KDSoapServerObjectInterface *serverO
         delete device;
         return true; // handled!
     }
-    const QByteArray response = httpResponseHeaders(false, contentType, device->size(), m_serverObject);
+
+    if (auto r = determineFileRanges(device->size()); !r) {
+        delete device;
+        return true;
+    } else if (const QVector<QPair<int, int>> requestedRanges = *r; !requestedRanges.isEmpty()) {
```

Comment:
This method is getting very long, could the body of this if() be moved to a separate method?